pkg/controller/testutils/testutils.go

```diff
@@ -32,3 +37,12 @@ func GetReconcileMetricCount(t *testing.T, controllerName, label string) float64
 	beforeCount := metricProto.GetCounter().GetValue()
 	return beforeCount
 }
+
+func StartTestingEnv() (*rest.Config, error) {
+	env = &envtest.Environment{}
```

Comment:
This is assigning env to a new envtest environment every time it's called. How do old ones get cleaned up?

Reply:
I'm not sure how much value having a utils for the testing env is bringing. Now we have to solve the problem of keeping state of each different test package env all in the same place? Doesn't seem worth it over just defining a envtest.Environment and starting and stopping in each package.

Reply:
Thought about it some more. We can have a singleton of testenv

```Go
var env *envtest.Environment
var usages int

func EnsureTestingEnv() (*rest.Config, error) {
   if env != nil {
       usages++
       return env.GetConfig(), nil // don't actually know the fn for getting the config for an env
   }

   new := &envtest.Environment{}
   cfg, err := env.Start()
   if err != nil {
      return nil, fmt.Errorf("starting environment: %w", err)
   }
   
   env = new
   usages++
   return cfg, nil
}

func ReleaseTestingEnv() error {
    usages--
    if usages < 1 {
        return env.Stop()
    }
   
    return nil
}
```

This would be the general strategy but we'd need to use either the atomic or sync packages to make this thread safe.

Reply:
This could exist in utils and solve all the problems.

Reply:
Solved by added envtest variable to start/stop functions:
```
func StartTestingEnv() (*rest.Config, *envtest.Environment, error) {
	env = &envtest.Environment{}
	restConfig, err := env.Start()
	if err != nil {
		return nil, nil, err
	}
	return restConfig, env, nil
}

func CleanupTestingEnv(env *envtest.Environment) error {
	return env.Stop()
}
```